### PR TITLE
Make response expiry date mandatory

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?
Payload format send from RM is being updated. `response_expires_at` is now a mandatory field,  validated at launch. Defaulting to None or datetime has been removed.
Related schema definitions PR -> https://github.com/ONSdigital/ons-schema-definitions/pull/83

### How to review
There are unit, integration and functional tests that required refactoring for this to work. Check if any defaulting to a value hasn't been omitted.

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by petechd** [comment]
[Make response expiry date mandatory for eQ Runner (1 SP)](https://trello.com/c/DiDKxlW6/5676-make-response-expiry-date-mandatory-for-eq-runner-1-sp)

File: tests/integration/create_token.py [hunk 1]
```diff
@@ -88,6 +88,7 @@ def _get_payload_with_params(
         payload_vars["exp"] = payload_vars["iat"] + float(3600)  # one hour from now
         payload_vars["jti"] = str(uuid4())
         payload_vars["case_id"] = str(uuid4())
+        payload_vars["response_expires_at"] = "2023-05-18T10:38:13+00:00"
```
Comment (MebinAbraham) [inline tests/integration/create_token.py:91]: Will this work past 18th?. I suggest just setting where ever we need these to like 
```py
(datetime.now(tz=timezone.utc) + timedelta(days=1)).isoformat()
```

**Review by MebinAbraham — COMMENTED** [review]
There is still logic in runner to handle this not existing, such as in questionnaire store. They should be updated.

File: tests/integration/create_token.py (hunk 1 above)
Comment (petechd) [reply tests/integration/create_token.py:91]: I was wondering about that too, dynamic date is much better, thanks.

**Review by rmccar — APPROVED** [review]

File: tests/app/views/handlers/test_individual_response_fulfilment_request.py [hunk 2]
```diff
@@ -27,6 +27,7 @@ def test_sms_fulfilment_request_payload():
         response_id="response_id",
         account_service_url="account_service_url",
         collection_exercise_sid="collection_exercise_sid",
+        response_expires_at=datetime.now(tz=timezone.utc),
```
[the remark this answers is not in the document]
Comment (petechd) [reply tests/app/views/handlers/test_individual_response_fulfilment_request.py]: Yep, updated now.

File: tests/integration/create_token.py [hunk 3]
```diff
@@ -88,6 +89,9 @@ def _get_payload_with_params(
         payload_vars["exp"] = payload_vars["iat"] + float(3600)  # one hour from now
         payload_vars["jti"] = str(uuid4())
         payload_vars["case_id"] = str(uuid4())
+        payload_vars["response_expires_at"] = (
+            datetime.now(tz=timezone.utc) + timedelta(days=1)
+        ).isoformat()
```
[the remark this answers is not in the document]
Comment (petechd) [reply tests/integration/create_token.py]: Added, I am unsure about the name though.

[the remark this answers is not in the document]
Comment (petechd) [reply tests/integration/create_token.py]: I've put the helper method in parser tests as that's where it is invoked the most.

**Review by katie-gardner — CHANGES_REQUESTED** [review]
Now mandatory in questionnaire store and added where relevant - I think some of the tests are still missing this like `test_metadata_proxy.py` and the token creation in integration tests - udpate here too?

But checked that I cannot launch a survey without it 👍

**Comment by petechd** [comment]
> Now mandatory in questionnaire store and added where relevant - I think some of the tests are still missing this like `test_metadata_proxy.py` and the token creation in integration tests - udpate here too?
> 
> But checked that I cannot launch a survey without it 👍

I've added "response_expires_at" to metadata in quite a few more tests except individual response which is irrelevant now. In create_token the field is added when _get_payload_with_params is invoked, same way as "tx_id", have a look.

**Comment by katie-gardner** [comment]
> > Now mandatory in questionnaire store and added where relevant - I think some of the tests are still missing this like `test_metadata_proxy.py` and the token creation in integration tests - udpate here too?
> > But checked that I cannot launch a survey without it 👍
> 
> I've added "response_expires_at" to metadata in quite a few more tests except individual response which is irrelevant now (will be removed, not being used atm). In create_token this field is added when _get_payload_with_params is invoked, same way as "tx_id", have a look.

Makes sense - thanks for explaining 👍

**Review by katie-gardner — APPROVED** [review]

**Review by MebinAbraham — APPROVED** [review]

File: tests/app/data_model/test_metadata_proxy.py [hunk 4]
```diff
@@ -22,6 +23,7 @@
     "tx_id": "tx_id",
     "collection_exercise_sid": "collection_exercise_sid",
     "case_id": "case_id",
+    "response_expires_at": "2023-05-24T10:46:32+00:00",
```
[the remark this answers is not in the document]
Comment (petechd) [reply tests/app/data_model/test_metadata_proxy.py:26]: it compares two instances (that might be also created at slightly different time but I didn't dig into it) and they are not the same object so it errors in test.

[the remark this answers is not in the document]
Comment (petechd) [reply tests/app/data_model/test_metadata_proxy.py:26]: Cases an issue when we run simultaneous workers while testing, so I hardcoded the value:
```
Different tests were collected between gw0 and gw10. The difference is:
--- gw0

+++ gw10

@@ -249,7 +249,7 @@

 tests/app/data_model/test_metadata_proxy.py::test_metadata_proxy_returns_value_for_valid_key[432423423423-4324234234231]
 tests/app/data_model/test_metadata_proxy.py::test_metadata_proxy_returns_value_for_valid_key[1_0000-1_00000]
 tests/app/data_model/test_metadata_proxy.py::test_metadata_proxy_returns_value_for_valid_key[1_0000-1_00001]
-tests/app/data_model/test_metadata_proxy.py::test_metadata_proxy_returns_value_for_valid_key[2023-05-19T07:47:25.339069+00:00-2023-05-19T07:47:25.339069+00:00]
+tests/app/data_model/test_metadata_proxy.py::test_metadata_proxy_returns_value_for_valid_key[2023-05-19T07:47:26.848739+00:00-2023-05-19T07:47:26.848739+00:00]

```

[the remark this answers is not in the document]
Comment (petechd) [reply tests/app/data_model/test_metadata_proxy.py:26]: I tried making a copy of the method and invoking in this file and it still causes the same issue, so I will hardcode it to the past if you don't mind.

**Pull request merged**